### DISCUSSION TITLE
What are the sub-process, memory and CPU requirements of Outline ?

### DISCUSSION THREAD

**Discussion by johnoscott** [outline/outline#2725] [Self Hosting] [answered]
I would like to understand :

1. how many sub-processes are spawned when Outline starts up (and what do they do) ?
2. how much memory is required ?
3. how much CPU is needed ? (cores and normal load)

I have dedicated a Mac Mini M1 16GB as a development server to share changes to my remote team.

I was surprised at how much memory and sustained CPU load there is when I run :
- local postgres, redis and fake S3 in docker
- `yarn start` from a build version running node v14

See the usage from the screenshot 1 below:
- after `yarn start` and before any requests from the browser

![Screenshot 2021-11-03-at-20-55-14](https://user-images.githubusercontent.com/291958/140040325-175bff64-00a8-4943-a3c9-33df88d3bbef.png)

**Comment by tommoor** [comment, marked as answer]
Two things of note in your screenshot:

- By default Outline will run one thread per core, so you can see it running 8 threads here. You can dial this back with the `WEB_CONCURRENCY` env variable, eg set it to 1.
- The other major contributor is your screensharing software, running at 15% CPU

**Comment by johnoscott** [comment]
I thought the default for `WEB_CONCURRENCY` was 1 thread. Is it 8 ?
